axelrod/strategies/punisher.py

```diff
@@ -143,3 +143,42 @@ def strategy(self, opponent: Player) -> Action:
             return D
         else:
             return C
+
+
+class TrickyLevelPunisher(Player):
+    """
+    A player starts by cooperating however, after 10, 50 and 100 rounds
+    will defect if at any point the number of defections
+    by an opponent is greater than 20%, 10% and 5% respectively.
+
+    Names:
+
+    - Tricky Level Punisher: [Eckhart2015]_
+    """
+
+    name = 'Tricky Level Punisher'
+    classifier = {
+        'memory_depth': float('inf'), # Long Memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def strategy(self, opponent: Player) -> Action:
+        if len(opponent.history) < 10:
+            return C
+        if opponent.defections / (opponent.defections + opponent.cooperations) > 0.2:
```

Comment:
For the sake of readability could we write this like this::

```

if len(opponent.history) < 50:
    if opponent.defections / len(opponent.history) > 0.2:
        return D
    return C
if len(opponent.history) < 100:
    if opponent.defections / len(opponent.history) > 0.1:
        return D
    return C
if opponent.defections / len(opponent.history) > 0.1:     
    return D
return C
```

This might be equivalent to what you have but it's more in line with the original (https://github.com/jecki/CoopSim/blob/master/Strategies.py#L1022) and I feel is also more readable.